src/controllers/controllerhidreporttabsmanager.cpp

```diff
@@ -0,0 +1,498 @@
+    if (reportType != hid::reportDescriptor::HidReportType::Input) {
+        pTable->setItemDelegateForColumn(5, new ValueItemDelegate(pTable));
+    }
+
+    bool showVolatileColumn = (reportType == hid::reportDescriptor::HidReportType::Feature ||
+            reportType == hid::reportDescriptor::HidReportType::Output);
+
+    // Set headers
+    QStringList headers = {QStringLiteral("Byte Position"),
+            QStringLiteral("Bit Position"),
+            QStringLiteral("Bit Size"),
+            QStringLiteral("Logical Min"),
+            QStringLiteral("Logical Max"),
+            QStringLiteral("Value"),
+            QStringLiteral("Physical Min"),
+            QStringLiteral("Physical Max"),
+            QStringLiteral("Unit Scaling"),
+            QStringLiteral("Unit"),
+            QStringLiteral("Abs/Rel"),
+            QStringLiteral("Wrap"),
+            QStringLiteral("Linear"),
+            QStringLiteral("Preferred"),
+            QStringLiteral("Null")};
+    if (showVolatileColumn) {
+        headers << QStringLiteral("Volatile");
+    }
+    headers << QStringLiteral("Usage Page") << QStringLiteral("Usage");
+
+    pTable->setColumnCount(headers.size());
+    pTable->setHorizontalHeaderLabels(headers);
+    pTable->verticalHeader()->setVisible(false);
+
+    // Helpers
+    auto createReadOnlyItem = [](const QString& text, bool rightAlign = false) {
+        auto* item = new QTableWidgetItem(text);
+        item->setFlags(item->flags() & ~Qt::ItemIsEditable);
+        if (rightAlign) {
+            item->setTextAlignment(Qt::AlignRight | Qt::AlignVCenter);
+        }
+        return item;
+    };
+    auto createValueItem = [reportType](int minVal, int maxVal) {
+        auto* item = new QTableWidgetItem;
+        QFont font = item->font();
+        font.setBold(true);
+        item->setFont(font);
+        if (reportType == hid::reportDescriptor::HidReportType::Input) {
+            item->setFlags(item->flags() & ~Qt::ItemIsEditable);
+        } else {
+            item->setFlags(item->flags() | Qt::ItemIsEditable);
+            item->setData(Qt::UserRole, QVariant::fromValue(QPair<int, int>(minVal, maxVal)));
+        }
+        return item;
+    };
+
+    int row = 0;
+    for (const auto& pControl : controls) {
+        // Column 0 - Byte Position
+        auto* bytePositionItem = createReadOnlyItem(QStringLiteral("0x%1").arg(QString::number(
+                                                            pControl.m_bytePosition, 16)
+                                                                    .rightJustified(2, '0')
+                                                                    .toUpper()),
+                true);
+        pTable->setItem(row, 0, bytePositionItem);
+        // Store custom data for the row in the first cell
+        bytePositionItem->setData(Qt::UserRole + 1,
+                QVariant::fromValue(reinterpret_cast<void*>(
+                        const_cast<hid::reportDescriptor::Control*>(
```

Comment:
You can 
```
Q_DECLARE_METATYPE(const void*)
```
To use const void* in QVariant. 

or probably directly 
```
Q_DECLARE_METATYPE(const hid::reportDescriptor::Control*)
```
(not tested)

Reply:
:+1: for the second.